File: subprojects/com.toedter.chatty.client.web/src/main/ts/model/ChatMessage.ts
/**
 * Copyright (c) 2014 Kai Toedter
 * All rights reserved.
 * Licensed under MIT License, see http://toedter.mit-license.org/
 */

/// <reference path="../../../../typings/angularjs/angular-resource.d.ts" />

module chatty.model {
    export interface ChatMessage {
        id: number;
        author: string;
        text: string;
        timeStamp: string;
    }

    export interface ChatMessageResource extends ChatMessage, ng.resource.IResource<ChatMessage> {
    }

    export interface ChatMessagesResource extends ng.resource.IResourceClass<ChatMessageResource> {
    }
}

File: subprojects/com.toedter.chatty.model/src/main/java/com/toedter/chatty/model/ChatMessage.java
/**
 * Copyright (c) 2016 Kai Toedter
 * All rights reserved.
 * Licensed under MIT License, see http://toedter.mit-license.org/
 */

package com.toedter.chatty.model;

import java.time.Instant;

public interface ChatMessage {
    long getId();

    String getText();

    User getAuthor();

    Instant getTimeStamp();
}
